src/test/java/org/jitsi/jicofo/xmpp/ConferenceIqProviderTest.java

```diff
@@ -91,16 +91,22 @@ public void testParseConferenceIqWithWrongRoom()
                     "</conference>" +
                 "</iq>";
 
+        // we expect that an exception will be thrown
+        Exception resultException = null;
         try
         {
             ConferenceIqProvider provider = new ConferenceIqProvider();
             IQUtils.parse(iqXml, provider);
         }
         catch(Exception e)
         {
-            // we expect XmppStringprepException
-            assertEquals(XmppStringprepException.class, e.getClass());
-        }   
+            resultException = e;
```

Comment:
I think it's cleaner to do the catch specifically for `XmppStringprepException` and put a fail statement at line 100 after parse which is the case when exception is not thrown.

Reply:
and the fail message says "XmppStringprepException was not thrown as expected"

Reply:
But isn't it clearer this way, in case of the failure you will have:

[ERROR] Failures:
[ERROR]   ConferenceIqProviderTest.testParseConferenceIqWithWrongRoom:102 expected:<class org.jxmpp.stringprep.XmppStringprepException> but was:<class java.lang.ArrayIndexOutOfBoundsException>

While in the other case the test throwing an error ... and you need to find out why, while here it is written: expected is this, not that.

Reply:
it's just not a standard pattern for testing thrown exceptions

Reply:
🙄
```
shouldThrow<XmppStringprepException> {
    IQUtils.parse(iqXml, provider)
}
```